### PR TITLE
fix: deduplicate ConfigMap key if ENI mode and endpointRoutes are enabled

### LINKED ISSUE #31868: Helm chart generates duplicate ConfigMap value
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Enabling the `eni` integration and explicitly enabling `endpointRoutes` in the chart should be redundant, as `eni.enabled` will enable endpoint routing, but it looks like similar to #13999 the `enable-endpoint-routes` key in the resultant `ConfigMap` is not deduplicated.

Minimal reproduction:
```console
bash-5.2$ helm template cilium cilium/cilium --version 1.15.2 \
  --namespace cilium \
  --set endpointRoutes.enabled=true \
  --set eni.enabled=true > cilium-eni.yaml
bash-5.2$ yamllint -d relaxed --no-warnings cilium-eni.yaml
cilium-eni.yaml
  191:3     error    duplication of key "enable-endpoint-routes" in mapping  (key-duplicates)
  793:1     error    trailing spaces  (trailing-spaces)
  809:1     error    trailing spaces  (trailing-spaces)
```

### Cilium Version

1.15.2

### Kernel Version

N/A

### Kubernetes Version

N/A

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by remi-gelinas**
Currently, if the Helm values `eni.enabled` and `endpointRoutes.enabled` are both explicitly enabled, the `enable-endpoint-routes` key in the main `ConfigMap` will be duplicated and fail schema validation. This PR adds an explicit check for `endpointRoutes.enabled` when `eni.enabled` is true to avoid duplicating the key, as discussed in #31868.

Fixes: #31868

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-configmap.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: Unless I'm missing something, I believe the `and` is not necessary here: Helm should correctly parse `if not .Values.endpointRoutes.enabled` correctly even if `.Values.endpointRoutes` is unset

**Review by gandro — COMMENTED** [review]
Great, thank you! One minor nit, otherwise this looks good to me

File: install/kubernetes/cilium/templates/cilium-configmap.yaml
Comment (remi-gelinas) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: Yup, you're right! Validated via local template, the main object check is unnecessary. Fixed!

**Review by gandro — APPROVED** [review]
Great, thank you!

**Comment by remi-gelinas** [comment]
@gandro Hey, just wanted to follow up on this - is there a reason image builds are failing on this branch when the only changes are to the Helm chart? Is there any action I need to take to get this mergeable?

**Comment by gandro** [comment]
> @gandro Hey, just wanted to follow up on this - is there a reason image builds are failing on this branch when the only changes are to the Helm chart? Is there any action I need to take to get this mergeable?

Let me restart the affected workflow, it might have been a flake.

As for next steps, there is nothing required from you at this point. We're still waiting on a review from sig-k8s. Once that is in (and assuming passing CI), we can merge then

**Comment by gandro** [comment]
@remi-gelinas Would you mind rebasing on main? It seems the build commit error is persistent, it might be because your branch is lagging behind a bit

**Review by aanm — APPROVED** [review]

**Pull request merged**
